### DISCUSSION TITLE
Cannot "see" manager DI on MAUI project

### DISCUSSION THREAD

**Discussion by AngeloLazzari** [shinyorg/shiny#1540] [Q&A]
Hi, we have all this nuget in our MAUI project
![image](https://github.com/user-attachments/assets/57c1eb91-7f09-412a-b3f0-9f3159b141ca)

We have the useShiny
![image](https://github.com/user-attachments/assets/70ad474f-1cc9-41ad-982b-7d69678a0498)

we put the addJobs but, when we have to set the job things up Visual Studio does not recognize the manager variable... 
![image](https://github.com/user-attachments/assets/d46bfd29-0cb8-44e5-8c85-6469deed13ab)

What are we missing? 

Thank you very much!

**Comment by AngeloLazzari** [comment]
The images are from our MauiProgram file...

Thank you

**Comment by aritchie** [comment]
Shiny uses dependency injection at the heart of everything it does.  My suggestion is to go read documentation on how that works in the MAUI repo as this question depends on a) how you want to do things, b) you're calling this before the service container is built, & c) far exceeds what I'm willing to help with on this forum

**Comment by AngeloLazzari** [comment]
Thank you. I do this because we are migrating from a old xamarin form to MAUI, and originally the job info was provided into the shinystartup, so I thought it was now the right place into the MauiProgram, and your documentation seem to do the same 

I will move the manager and things on the first time I need them so...

Thank you very much
